controllers/provider-azure/pkg/controller/controlplane/valuesprovider.go

```diff
@@ -209,6 +217,7 @@ func getConfigChartValues(
 		"routeTableName":    routeTableName,
 		"securityGroupName": securityGroupName,
 		"region":            cp.Spec.Region,
+		"maxNodes":          maxNodes,
```

Comment:
```suggestion
		"maxNodes":          fmt.Sprintf("%d", maxNodesCount)
```

Reply:
Why is `fmt.Sprintf` needed? Also, why do we need two variables `maxNodes` and `maxNodesCount`?

Reply:
Sorry, it was a dummy approach to fix unit tests failures.
They were failing because `int(0)!=int32(0)`, which I got too late.